### PR TITLE
feat: Added sonarcloud Github Action

### PR DESCRIPTION
**Pull request by pascalwilbrink**
feat: Added sonarcloud Github Action
fixes: #1242

### CODE REVIEW

**Comment by juancho0202** [comment]
@pascalwilbrink let's apply the same principle here of splitting the process into an unprivileged build workflow and a subsequent one triggered by `workflow_run` to have access to the Sonarcloud secret and analyze forked PRs

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+name: SonarCloud Analysis
+
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - 'main'
+      - 'develop'
```
Comment (Stef3st) [inline .github/workflows/sonarcloud.yml:14]: There's no develop branch on open-scd. This can be removed

File: .github/workflows/sonarcloud.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  pull_request_target:
+    branches:
+      - 'main'
+      - 'develop'
```
Comment (Stef3st) [inline .github/workflows/sonarcloud.yml:18]: There's no develop branch on open-scd. This can be removed

**Review by Stef3st — CHANGES_REQUESTED** [review]
Before implementing we might want to check what rules/boundries we'd like to apply here

**Comment by trusz** [comment]
Is this PR still relevant?

**Comment by trusz** [comment]
It is weird that sonarcloud did not find anything. We have to check the configs.

**Comment by juancho0202** [comment]
> It is weird that sonarcloud did not find anything. We have to check the configs.

@trusz it is weird because if you navigate to their analysis of the main branch there are 45 bugs detailed in there: https://sonarcloud.io/project/issues?resolved=false&id=openscd_open-scd

**Comment by trusz** [comment]
The SonarCloud projects does not exist anymore. I will close this PR, but we can reopen it again if we want to proceed with SonarCloud

**Pull request closed without merging**